### PR TITLE
Add lazy v2 call header reading

### PR DESCRIPTION
**Pull request by jcorbin**
So that the upcoming stats path can get such things as the `cn` header.

r @Raynos @kriskowal 

Cheers @abhinav for making me think to use a typed array.

### CODE REVIEW

File: v2/header.js [hunk 1]
```diff
@@ -229,3 +259,59 @@ module.exports.header2 = HeaderRW(bufrw.UInt16BE, bufrw.str2, bufrw.str2, {
...
+    if (this.index < this.data.length) {
+        this.data[this.index++] = keyOffset;
+        this.data[this.index++] = keyLength;
+        this.data[this.index++] = valOffset;
+        this.data[this.index++] = valLength;
+    }
+};
+
+KeyVals.prototype.getValue =
+function getValue(key) {
```
Comment (kriskowal) [inline v2/header.js]: aka get

File: v2/header.js [hunk 2]
```diff
@@ -229,3 +259,59 @@ module.exports.header2 = HeaderRW(bufrw.UInt16BE, bufrw.str2, bufrw.str2, {
     maxHeaderCount: Infinity,
     maxKeyLength: Infinity
 });
+
+function KeyVals(buffer, length) {
```
Comment (kriskowal) [inline v2/header.js]: aka lazymap

**Comment by kriskowal** [comment]
:ok:

File: v2/header.js [hunk 3]
```diff
@@ -229,3 +259,59 @@ module.exports.header2 = HeaderRW(bufrw.UInt16BE, bufrw.str2, bufrw.str2, {
...
+        }
+
+        var keyOffset = this.data[i];
+        var found = true;
+        for (
+            var j = 0, offset = keyOffset;
+            j < keyLength;
+            j++, offset++
+        ) {
+            if (key[j] !== this.buffer[offset]) {
```
Comment (Raynos) [inline v2/header.js]: How much sad when using arbitrary UTF8 surrogate pairs ?

File: v2/header.js (hunk 2 above)
Comment (Raynos) [inline v2/header.js]: can you write a block comment on how this data structure works ?

Would be nice for future readers.

Or a TODO.

**Pull request merged**